canonical-web-and-design/maas-ui
# refactor(legacy): clean up nodes list code

## Done

- Remove machine and pools code from the nodes list. 
- Fix updating the selected controllers. 
- Fix the script selector for controllers.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

You might need sampledata to get more controllers for these tests.
- Visit the controllers list.
- Select all controllers.
- Open the take action menu and select an action that no all controllers can do.
- It should show you the link to update your selection. Click it.
- The selection should update and the form should appear.
- Cancel and open the test form.
- Try changing the selected tests. You should be able to add and remove them.
- Visit the devices list, you should be able to perform actions as normal.

## Fixes

Fixes: #2503.
